tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/AutoClientIDShardClusterTest.java

```diff
@@ -0,0 +1,343 @@
+
+      DurableSub(String id) {
+         this.id = id;
+      }
+
+      @Override
+      public void run() {
+         while (!consumerDone.get()) {
+            try {
+               ConnectionFactory connectionFactory = createFactory(protocol, "ClientId-" + id, "admin", "admin");
+               try (Connection connection = connectionFactory.createConnection()) {
+                  connection.start();
+                  try (Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)) {
+                     javax.jms.Topic topic = session.createTopic("T");
+                     try (TopicSubscriber durableSubscriber = session.createDurableSubscriber(topic, "Sub-" + id)) {
+                        registered.countDown();
+                        for (int i = 0; i < 5; i++) {
+                           Message message = durableSubscriber.receive(500);
+                           if (message != null) {
+                              lastReceived = message.getIntProperty("SEQ");
+                              if (lastReceived > maxReceived) {
+                                 maxReceived = lastReceived;
+                              }
+                              if (receivedInOrder < 0) {
+                                 receivedInOrder = lastReceived;
+                              } else if (receivedInOrder == lastReceived - 1) {
+                                 receivedInOrder++;
+                              } else {
+                                 if (!orderShot.get()) {
+                                    System.err.println("Sub: " + id + ", received: out of order " + lastReceived + ", last in order: " + receivedInOrder);
+                                 }
+                                 orderShot.set(true);
+                              }
+                           } else {
+                              // no point trying again if there is nothing for us now.
+                              break;
+                           }
+                        }
+                        TimeUnit.MILLISECONDS.sleep(500);
+                     }
+                  }
+               }
+            } catch (Exception ok) {
+            }
+         }
+      }
+   }
+
+   @Ignore("not totally reliable, but does show the root cause of the problem being solved")
+   public void testWithoutOutSharding() throws Exception {
+      setupServers();
+      startServers(0, 1);
+
+      // two bouncy durable consumers
+      DurableSub sub0 = new DurableSub("0");
+      DurableSub sub1 = new DurableSub("1");
+
+      new Thread(sub0).start();
+      new Thread(sub1).start();
+
+      // waiting for registration before production to give bridges a chance
+      assertTrue(sub0.registered.await(20, TimeUnit.SECONDS));
+      assertTrue(sub1.registered.await(20, TimeUnit.SECONDS));
+
+
+      // wait for bindings!
+      assertTrue(Wait.waitFor(() -> 2 == gatherBindingsOnT(0)[0], 20000, 2000));
```

Comment:
is it possible to reuse ClusterTestBase.waitForBindings?

Reply:
hmm, just tried this, and wait for bindings has a consumer count expectation, that I don't want b/c my consumers are bouncing!

Reply:
I guess I can add support for -1 consumerCount